folx/experimental/pallas/utils.py

```diff
@@ -0,0 +1,161 @@
+from typing import Optional, Tuple
```

Comment:
These utils are very specific to mha which is not clear from the file path. Could we move everything into a `mha` submodule?